src/main/java/uk/co/real_logic/benchmarks/rtt/Harness.java

```diff
@@ -0,0 +1,235 @@
+/*
+ * Copyright 2015-2020 Real Logic Limited.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.co.real_logic.benchmarks.rtt;
+
+import org.HdrHistogram.Histogram;
+import org.agrona.SystemUtil;
+import org.agrona.concurrent.IdleStrategy;
+import org.agrona.concurrent.NanoClock;
+import org.agrona.concurrent.SystemNanoClock;
+
+import java.io.PrintStream;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.atomic.AtomicLong;
+
+import static java.lang.Math.*;
+import static java.util.Objects.requireNonNull;
+import static java.util.concurrent.CompletableFuture.runAsync;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static java.util.stream.Collectors.toMap;
+
+/**
+ * {@code Harness} class is the core of the RTT benchmark. It is responsible for running benchmark against provided
+ * {@link MessageProvider} instance using given {@link Configuration}.
+ */
+public final class Harness
```

Comment:
Maybe LoadHarness would be more descriptive?

Reply:
Probably a more correct term is a "Rig". This would be a LoadTestRig.

Reply:
Renamed